### DISCUSSION TITLE
Is there anything like impose for static effects?

### DISCUSSION THREAD

**Discussion by mkleczek** [haskell-effectful/effectful#153] [Q&A]
In other words - is there any example of implementing one static effect using another?

**Comment by ocharles** [comment]
I don't believe this is possible because by definition a static effect is defined implicitly at each of its operations, rather than having a dedicated interpreter. Perhaps you could share more about what it is you want to do, to avoid an X/Y problem?

**Reply by mkleczek** [reply]
I reported https://github.com/haskell-effectful/effectful/issues/152 - I think current Error effect API is somewhat convoluted as `catch` does not discharge (so what's the point of catching if you have to deal with the error elsewhere anyway)

I tried to implement a simple Throws effect:
```
data Throws e :: Effect
```

Seeing how complex Error implementation is I wanted to try to implement `Throws` in terms of `Error` (not sure how yet).

**Reply by mkleczek** [reply]
> I don't believe this is possible because by definition a static effect is defined implicitly at each of its operations, rather than having a dedicated interpreter. Perhaps you could share more about what it is you want to do, to avoid an X/Y problem?

Well - it is possible in case of IOE thanks to unsafeEff.
